CompilerSource/compiler/compile.cpp

```diff
@@ -602,30 +601,25 @@ wto << "namespace enigma_user {\nstring shader_get_name(int i) {\n switch (i) {\
 
   idpr("Starting compile (This may take a while...)", 30);
 
-  string make = MAKE_flags;
+  string make = compiler_settings["MAKEFLAGS"];
 
-  make += " Game ";
+  make += "Game ";
   make += "WORKDIR=\"" + eobjs_directory + "\" ";
   make += "CODEGEN=\"" + codegen_directory + "\" ";
-  make += mode == emode_debug? "GMODE=Debug ": mode == emode_design? "GMODE=Design ": mode == emode_compile?"GMODE=Compile ": "GMODE=Run ";
-  make += "GRAPHICS=" + extensions::targetAPI.graphicsSys + " ";
-  make += "AUDIO=" + extensions::targetAPI.audioSys + " ";
-  make += "COLLISION=" + extensions::targetAPI.collisionSys + " ";
-  make += "WIDGETS="  + extensions::targetAPI.widgetSys + " ";
-  make += "NETWORKING="  + extensions::targetAPI.networkSys + " ";
-  make += "PLATFORM=" + extensions::targetAPI.windowSys + " ";
-
-  if (CXX_override.length()) make += "CXX=" + CXX_override + " ";
-  if (CC_override.length()) make += "CC=" + CC_override + " ";
-  if (WINDRES_location.length()) make += "WINDRES=" + WINDRES_location + " ";
-
-  if (TOPLEVEL_cflags.length()) make += "CFLAGS=\"" + TOPLEVEL_cflags + "\" ";
-  if (TOPLEVEL_cppflags.length()) make += "CPPFLAGS=\"" + TOPLEVEL_cppflags + "\" ";
-  if (TOPLEVEL_cxxflags.length()) make += "CXXFLAGS=\"" + TOPLEVEL_cxxflags + "\" ";
-  if (TOPLEVEL_ldflags.length()) make += "LDFLAGS=\"" + TOPLEVEL_ldflags + "\" ";
-  if (TOPLEVEL_ldlibs.length()) make += "LDLIBS=\"" + TOPLEVEL_ldlibs + "\" ";
-
-  if (TOPLEVEL_rcflags.length()) make += "RCFLAGS=\"" + TOPLEVEL_rcflags + "\" ";
+  make += mode == emode_debug? "GMODE=\"Debug\"" : mode == emode_design? "GMODE=\"Design\"" : mode == emode_compile?"GMODE=\"Compile\"" : "GMODE=\"Run\"";
+  make += " ";
+  make += "GRAPHICS=\"" + extensions::targetAPI.graphicsSys + "\" ";
+  make += "AUDIO=\"" + extensions::targetAPI.audioSys + "\" ";
+  make += "COLLISION=\"" + extensions::targetAPI.collisionSys + "\" ";
+  make += "WIDGETS=\""  + extensions::targetAPI.widgetSys + "\" ";
+  make += "NETWORKING=\""  + extensions::targetAPI.networkSys + "\" ";
+  make += "PLATFORM=\"" + extensions::targetAPI.windowSys + "\" ";
+
+  std::string ey_keys[] = {"NAME", "NATIVE", "MAINTAINER", "TARGET-PLATFORM", "CXX", "CC", "CFLAGS", "CXXFLAGS", "CPPFLAGS", "WINDRES", "LDFLAGS", "LDLIBS", "RCFLAGS"};
```

Comment:
I think this list doesn't belong here... it may belong in `gcc_backend`, or it may belong in a new options management struct. I'm not opposed to either. I'm leaning toward the latter. When you deleted that huge block of variable assignments, you deleted the only compiler-checked collection of these options. We don't have a home base for options management, anymore (not that the list you deleted was a great example of one). This new list of options can only be used in a certain way—passed directly through to `make`. We need a place to declare options that can be used in other ways.